src/algebraic_geometry/is_open_comap_C.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebraic_geometry.prime_spectrum
+import ring_theory.polynomial.basic
+/-!
+The morphism `Spec R[x] --> Spec R` induced by the natural inclusion `R --> R[x]` is an open map.
+-/
+
+open ideal polynomial prime_spectrum set integral_domain
+
+local attribute [instance] classical.prop_decidable
+
+variables {R : Type*} [comm_ring R] {P : ideal R}
+
+/-- If `P` is a prime ideal of `R`, then `R[x]/(P)` is an integral domain. -/
+lemma is_integral_domain_C_map_quotient (H : is_prime P) :
+  is_integral_domain (quotient (map C P : ideal (polynomial R))) :=
+ring_equiv.is_integral_domain (polynomial (quotient P))
+  (to_is_integral_domain (polynomial (quotient P)))
+  (polynomial_quotient_equiv_quotient_polynomial P).symm
+
+/-- If `P` is a prime ideal of `R`, then `P.R[x]` is a prime ideal of `R[x]`. -/
+lemma is_prime_C_map_of_is_prime (H : is_prime P) : is_prime (map C P : ideal (polynomial R)) :=
+(quotient.is_integral_domain_iff_prime (map C P : ideal (polynomial R))).mp
+  (is_integral_domain_C_map_quotient H)
+
+/-- Given a polynomial `f ∈ R[x]`, `image_of_Df` is the subset of `Spec R` where at least one
+of the coefficients of `f` does not vanish.  We will prove that `image_of_Df` is the image of
+`(zero_locus {f})ᶜ` under the morphism `comap C : Spec R[x] → Spec R`. -/
+def image_of_Df (f : polynomial R) := {p : prime_spectrum R | ∃ i : ℕ , (coeff f i) ∉ p.1}
+
+lemma std_aff_is_open (a : R) : is_open ({p : prime_spectrum R | a ∉ p.1 }) :=
+begin
+  refine ⟨{a} , ext (λ x, ⟨λ xh, _,λ xh,  _⟩)⟩;
+  simpa only [mem_zero_locus, not_not, mem_set_of_eq, singleton_subset_iff, mem_compl_eq] using xh,
+end
+
+lemma is_open_image_of_Df (f : polynomial R) : is_open (image_of_Df f) :=
+begin
+  rw [image_of_Df, set_of_exists (λ i (x : prime_spectrum R), coeff f i ∉ x.val)],
+  exact is_open_Union (λ i, std_aff_is_open (coeff f i)),
+end
+
+lemma mem_span_C_coeff (f : polynomial R) :
+  f ∈ span {tt : polynomial R | ∃ i : ℕ ,tt = (C (coeff f i))} :=
```

Comment:
```suggestion
  f ∈ span {g : polynomial R | ∃ i : ℕ, g = (C (coeff f i))} :=
```
this should also move to `polynomial/something.lean`

Reply:
I think this should move to another file.
Also `tt` is used for the "true" value of `bool`. I've never seen it used as name for a polynomial. Would you mind changing it to `g` or `phi` or something?